### PR TITLE
#652 v4.0.0

### LINKED ISSUE #652: epic: release version 4.0.0
some tasks to do for release of 4.0.0:
* remove deprecated code
* remove `RandomIdOf` and `RandomId`
* remove `KafkaHealthCheck`
* #590
* #589 
* #638 
* #639
* #648 
* #653 
* update `circe` to 0.14.7 or later
* update `smetrics` 2.2.0 or later
* review and deal with all places, where `@nowarn` is used

### PR DESCRIPTION
**Pull request by mr-git**
Fixes #652

### CODE REVIEW

**Review by dfakhritdinov — APPROVED** [review]

File: eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/Schema.scala [hunk 1]
```diff
@@ -12,15 +12,13 @@ import com.evolutiongaming.scassandra.TableName
...
  * @param metaJournal stores information about active aggregates, see [[MetaJournalStatements]]
  * @param pointer is not used any more, deprecated and scheduled for removal, see [[PointerStatements]]
  * @param pointer2 stores information on how close to tail the replication is, see [[Pointer2Statements]]
  * @param setting used to track schema migrations, see [[SetupSchema]] and [[MigrateSchema]] for details
  */
 final case class Schema(
   journal: TableName,
-  metadata: TableName, // is not used // TODO MR remove with next major release
   metaJournal: TableName,
   pointer: TableName, // TODO MR remove with next major release
```
Comment (Phill101) [inline eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/Schema.scala:23]: What about this field?

Comment (Phill101) [reply eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/Schema.scala:23]: And table in general

Comment (mr-git) [reply eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/Schema.scala:23]: we wanted, but we may not rush till we have proof that nobody is relying on it any more

**Review by Phill101 — APPROVED** [review]

**Comment by mr-git** [comment]
Approved from my side, though I cannot express it as action as I am author

**Pull request merged**